src/index.js

```diff
@@ -49,9 +50,13 @@ class CID {
    * new CID(<cid.buffer>)
    * new CID(<multihash>)
    * new CID(<bs58 encoded multihash>)
+   * new CID(<CID>)
    *
    */
   constructor (version, codec, multihash) {
+    if (CID.isCID(version)) {
+      return version
```

Comment:
@enricomarino almost. you need to clone those properties, otherwise things like the multihash (Buffer, moved around by reference and not by value) will mutate in one CID if you mutate in the other, causing all sorts of strange errors.